rfcs/block-component.md

```diff
@@ -52,6 +52,59 @@ Props would be exposed for:
 
 > **Note** that we would likely _not_ use this component internally to build other baseui components. It's primarily a helper component for customers when building their apps.
```

Comment:
The above content can probably be moved down into some sort of "Motivation" section now, I'd imagine the API docs / usage examples are probably most useful.

Reply:
I've reordered the page and removed some of the unnecessary content.